### PR TITLE
Remove examples directory and improvement documentation

### LINKED ISSUE #25: The example directory seems useless?
Currently we have
 - `symbiflow-examples/examples/xc7`

The extra `examples` directory doesn't seem to add much value?

It would probably be better to have the following
 - `symbiflow-examples/xc7`

It also means that a person who goes to this repo quickly sees which architectures there are examples for.

### PR DESCRIPTION
**Pull request by kamilrakoczy**
Fixes #25
Supersede of #39 

CI will fail, as tests are downloading master version, not version from this PR.

This pull request:

- Removes the extra examples directory.
- Makes the layout of the eos-s3 and xc7 directories consistent.
- Moves the setup instructions + build instructions under the relevant locations.
- Makes the setup instructions for eos-s3 and xc7 consistent (install conda then create an environment using the correct environment.yml file).
- Sections for the eos-s3 and xc7 are links to the README sections in each directory. It is not possible to automatically generate content from another README.md file (see: https://stackoverflow.com/questions/35080160/github-include-md-files-in-readme-md).

### CODE REVIEW

**Comment by mithro** [comment]
> CI will fail, as tests are downloading master version, not version from this PR.

Can you log a github issue about this?

**Comment by mithro** [comment]
> * Sections for the eos-s3 and xc7 are links to the README sections in each directory. It is not possible to automatically generate content from another README.md file (see: https://stackoverflow.com/questions/35080160/github-include-md-files-in-readme-md).

See the solution we use in skywater-pdk -> https://github.com/google/skywater-pdk

File: eos-s3/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+## Setting up the toolchain
+
+```bash
+export INSTALL_DIR="/opt/symbiflow/eos-s3"
+bash conda_installer.sh -b -p $INSTALL_DIR/conda && rm conda_installer.sh
+source "$INSTALL_DIR/conda/etc/profile.d/conda.sh"
+conda env create -f eos-s3/environment.yml
+conda activate eos-s3
+wget -qO- https://quicklogic-my.sharepoint.com/:u:/p/kkumar/Eb7341Bq-XRAukVQ6oQ6PrgB-qdFbrsrPEON1yTa4krFSA?download=1 | tar -xJ -C $INSTALL_DIR
```
Comment (mithro) [inline eos-s3/README.md:13]: These two steps highlight that the quicklogic install instructions are wrong. They should not be referencing sharepoint / kkumar not doing a copy.

File: xc7/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+source "$INSTALL_DIR/conda/etc/profile.d/conda.sh"
+conda env create -f xc7/environment.yml
+conda activate xc7
+wget -qO- https://storage.googleapis.com/symbiflow-arch-defs/artifacts/prod/foss-fpga-tools/symbiflow-arch-defs/presubmit/install/477/20200714-082108/symbiflow-arch-defs-install-8eb88e76.tar.xz | tar -xJ --one-top-level=$INSTALL_DIR/install
+conda deactivate
+```
+
+## Building the examples
+
+To build the example, run the following commands:
```
Comment (mithro) [inline xc7/README.md:24]: Please have seperate instructions for each example.

File: eos-s3/README.md (hunk 1 above)
Comment (kamilrakoczy) [reply eos-s3/README.md:13]: This commands where copied directly from hidden installation script of [quicklogic toolchain installator](https://github.com/QuickLogic-Corp/quicklogic-fpga-toolchain/releases/download/v1.1.0/Symbiflow_v1.1.0.gz.run). 

To not reference sharepoint / kkumar quicklogic archdef needs to be uploaded somewhere else. Probably this should be done by someone from QL.

Copy is fixing the problem with yosys not finding shared files. It is probably a bug in the quicklogic-yosys conda package. The same copy is done in the hidden installation script.
If we do not want to directly copy we need to wait for a fixed conda package or still use the official installation script.

**Comment by kamilrakoczy** [comment]
> > CI will fail, as tests are downloading master version, not version from this PR.
> 
> Can you log a github issue about this?

I created new issue: #48

File: eos-s3/README.md (hunk 1 above)
Comment (mithro) [reply eos-s3/README.md]: > This commands where copied directly from hidden installation script of [quicklogic toolchain installator](https://github.com/QuickLogic-Corp/quicklogic-fpga-toolchain/releases/download/v1.1.0/Symbiflow_v1.1.0.gz.run).
> 
> To not reference sharepoint / kkumar quicklogic archdef needs to be uploaded somewhere else. Probably this should be done by someone from QL.

This should be generated from CI automatically. Talk to @kgugala about who should own fixing this.

> Copy is fixing the problem with yosys not finding shared files. It is probably a bug in the quicklogic-yosys conda package. The same copy is done in the hidden installation script.
> If we do not want to directly copy we need a fixed conda package or still use the official installation script.

Please log bugs about these issues. Both the conda package and the official installation script probably need fixing.

**Comment by mithro** [comment]
I'm going to merge this. There are a bunch of follow up items which need to be logged as issues and then cleaned up.

**Pull request merged**